### ISSUE TITLE
propogation of custom config from top to children

### ISSUE THREAD

**Issue by danziamo** [drom/resch#47]
right now config looks following way
```js
this.Form = genForm({
    schema: rockSchema,
    path: [],
    updateState: this.updateState,
});
```
It would be nice if resch could propogate down some custom config objects down (so there would be no necessity in creating custom widgets)
```
this.Form = genForm({
    schema: rockSchema,
    path: [],
    fn: {
        updateState: this.updateState,
        myCustomFunction: this.firstFn,
        anotherFn: this.secondFn
});
```

**Comment by danziamo** [comment]
If i want to host two resch applications using only one state. Is it possible to make such all data from 1st application would be stored under `data`, and all data from 2nd applications would be stored under other name, for example `parameters`. So it would be
```js
this.state = {
  data: undefined, // all data belonging to 1st resch
  parameters: undefined // all data belonging to 2nd resch
}
```
Cause right now i might have to duplicate default widgets

**Comment by drom** [comment]
@danziamo what if you keep `data` over one more layer?
like `state.app1.data` and `state.app2.data` ?

**Comment by danziamo** [comment]
@drom good idea. Will do this. It means. I just need to create another updateState function
